stencil_benchmarks/tools/compilation.py

```diff
@@ -126,8 +127,12 @@ def __init__(self,
         else:
             compile_command += ['-shared', '-fPIC']
 
-        with tempfile.NamedTemporaryFile(suffix=extension) as srcfile:
-            srcfile.write(code.encode())
+        output_dir = "benchmarks_source_code"
+        os.makedirs(output_dir, exist_ok=True)
+        file_path = os.path.join(output_dir, "{}{}".format(filename, extension))
+
+        with open(file_path, 'w') as srcfile:
+            srcfile.write(code)
```

Comment:
What about something like the following?
```suggestion
        output_dir = pathlib.Path("benchmarks_source_code")
        output_dir.mkdir(exist_ok=True)

        with tempfile.NamedTemporaryFile(suffix=extension, dir=output_dir, delete=False) as srcfile:
            srcfile.write(code.encode())
```
This allows for source inspection (as used by NCU), but avoids the problem of possibly overwriting existing source files with different ones (which might make NCU showing the wrong source for some results).

Reply:
Good point 👍 I also added the filename as a prefix to make it more human readable